jsonb-generator/src/main/java/io/avaje/jsonb/generator/TypeReader.java

```diff
@@ -263,18 +264,49 @@ private void readMethod(Element element, List<FieldReader> localFields) {
       }
     }
     // for getter/accessor methods only, not setters
-    PropertyPrism.getOptionalOn(methodElement).ifPresent(propertyPrism -> {
-      if (!methodElement.getParameters().isEmpty()) {
-        logError(errorContext + baseType + ", @Json.Property can only be placed on Getter Methods, but on %s", methodElement);
-        return;
-      }
-
-      // getter property as simulated read-only field with getter method
-      final var frequency = frequency(propertyPrism.value());
-      final var reader = new FieldReader(element, namingConvention, currentSubType, genericTypeParams, frequency);
-      reader.getterMethod(new MethodReader(methodElement));
-      localFields.add(reader);
-    });
+    PropertyPrism.getOptionalOn(methodElement)
+        .filter(p -> !hasRecordPropertyAnnotation(methodElement))
+        .ifPresent(
+            propertyPrism -> {
+              if (!methodElement.getParameters().isEmpty()) {
+                logError(
+                    errorContext
+                        + baseType
+                        + ", @Json.Property can only be placed on Getter Methods, but on %s",
+                    methodElement);
+                return;
+              }
+
+              // getter property as simulated read-only field with getter method
+              final var frequency = frequency(propertyPrism.value());
+              final var reader =
+                  new FieldReader(
+                      element, namingConvention, currentSubType, genericTypeParams, frequency);
+              reader.getterMethod(new MethodReader(methodElement));
+              localFields.add(reader);
+            });
+  }
+  
+  private boolean hasRecordPropertyAnnotation(ExecutableElement methodElement) {
+
+    try {
+      return APContext.jdkVersion() < 16
```

Comment:
Less than JDK 16 can't have records, so I'm misreading this ... jdk 11 returns true for hasRecordPropertyAnnotation()?

Reply:
We're checking if a recordproperty exists in this method, then doing `!` on the result in the filter